Remove use_c10_dispatcher: unboxed_only

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36850 Remove callBoxedWorkaround
* **#36838 Remove use_c10_dispatcher: unboxed_only**

All ops now do unboxing after dispatch, i.e. c10 handles unboxing and c10 registers a wrapper for the op to JIT
The last op that manually registered its own wrapper to JIT in register_aten_ops.cpp was migrated.

Since there are no ops using use_c10_dispatcher: unboxed_only anymore, we can delete the feature.

Also:
- Rename some files to more accurately describe what they're doing now:
  - OpsAlreadyMovedToC10.h/cpp -> ATenOpList.h/cpp
  - register_aten_ops.cpp -> generated_unboxing_wrappers.cpp
  - gen_jit_dispatch.py -> gen_unboxing_wrappers.py

Differential Revision: [D21100081](https://our.internmc.facebook.com/intern/diff/D21100081/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21100081/)!

smessmer committed:
Remove use_c10_dispatcher: unboxed_only

All ops now do unboxing after dispatch, i.e. c10 handles unboxing and c10 registers a wrapper for the op to JIT
The last op that manually registered its own wrapper to JIT in register_aten_ops.cpp was migrated.

Since there are no ops using use_c10_dispatcher: unboxed_only anymore, we can delete the feature.

Also:
- Rename some files to more accurately describe what they're doing now:
  - OpsAlreadyMovedToC10.h/cpp -> ATenOpList.h/cpp
  - register_aten_ops.cpp -> generated_unboxing_wrappers.cpp
  - gen_jit_dispatch.py -> gen_unboxing_wrappers.cpp

Differential Revision: [D21100081](https://our.internmc.facebook.com/intern/diff/D21100081/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21100081/)!

[ghstack-poisoned]

ezyang commented:
Do we still need `impl_UNBOXED`?

smessmer commented:
> Do we still need `impl_UNBOXED`?

Yes, to get rid of that, we first need to fix all the gaps in https://github.com/pytorch/pytorch/issues/32366.

ezyang commented on aten/src/ATen/core/ATenOpList.h:
```diff
@@ -0,0 +1,14 @@
+namespace at {
+
+// list of ATen ops that come from native_functions.yaml
+CAFFE2_API bool is_aten_op(const c10::OperatorName& opName);
```
This new name doesn't seem very descriptive. (A better question to ask, perhaps: why do I care if it `is_aten_op`? Why isn't this the same thing as just checking the namespace?)

bhosmer replied:
Agreed, the implications of this name aren't that clear unless you're pretty deep into what's going on. Judging by its use [here](https://github.com/pytorch/pytorch/pull/36838/files#diff-94205457152f6821c30329a4ddb40bbeR172), I'd suggest maybe inverting the sense and making it `is_custom_op` instead.

smessmer replied:
checking the namespace is a good idea, we should move towards doing that instead. I added a TODO. It likely doesn't work just yet because we might have internal "custom" ops that use the aten namespace.

ezyang replied:
@smessmer when I get everything onto the new API, we will be able to guarantee that custom ops don't use the aten namespace.

But I am kind of curious what the downstream usage of this is, because the way I'm likely going to make the internal "Custom" ops (you're talking about the JIT ops right?) live in a centralized location is by hardcoding their registrations in TypeDefault.cpp ala https://github.com/pytorch/pytorch/issues/36925

See also https://github.com/pytorch/pytorch/issues/36926

ezyang reviewed (approved):
This seems like a strict improvement (so I will approve) but I don't see why you need the `is_aten_op` predicate anymore.

bhosmer commented on torch/csrc/jit/runtime/register_c10_ops.cpp:
```diff
@@ -169,16 +169,11 @@ class RegistrationListener final : public c10::OpRegistrationListener {
     }
-    if (at::is_aten_op_and_unboxing_is_already_handled_by_c10(
-            op.schema().operator_name())) {
+    if (at::is_aten_op(op.schema().operator_name())) {
```
Aside: in a world where the listener setup is no longer needed (because the JIT knows how to do c10 registry lookups directly), where will this conditional creation of a testing wrapper wind up?

smessmer replied:
Excellent question, moving this to do a namespace check instead might solve the issue.

bhosmer reviewed (approved):
If @ezyang is right and `is_aten_op` can be replaced by a simple namespace check at the remaining callsites, that definitely seems better. Otherwise I'd vote for renaming per inline comment.

Also, not for this PR but in general I'd mildly prefer having the `unboxed_only` removal and the renaming be separate diffs, esp given that the file renaming ripples into the build scripts and so on.

smessmer committed:
Update on "Remove use_c10_dispatcher: unboxed_only"


All ops now do unboxing after dispatch, i.e. c10 handles unboxing and c10 registers a wrapper for the op to JIT
The last op that manually registered its own wrapper to JIT in register_aten_ops.cpp was migrated.

Since there are no ops using use_c10_dispatcher: unboxed_only anymore, we can delete the feature.

Also:
- Rename some files to more accurately describe what they're doing now:
  - OpsAlreadyMovedToC10.h/cpp -> ATenOpList.h/cpp
  - register_aten_ops.cpp -> generated_unboxing_wrappers.cpp
  - gen_jit_dispatch.py -> gen_unboxing_wrappers.cpp

Differential Revision: [D21100081](https://our.internmc.facebook.com/intern/diff/D21100081/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21100081/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove use_c10_dispatcher: unboxed_only"


All ops now do unboxing after dispatch, i.e. c10 handles unboxing and c10 registers a wrapper for the op to JIT
The last op that manually registered its own wrapper to JIT in register_aten_ops.cpp was migrated.

Since there are no ops using use_c10_dispatcher: unboxed_only anymore, we can delete the feature.

Also:
- Rename some files to more accurately describe what they're doing now:
  - OpsAlreadyMovedToC10.h/cpp -> ATenOpList.h/cpp
  - register_aten_ops.cpp -> generated_unboxing_wrappers.cpp
  - gen_jit_dispatch.py -> gen_unboxing_wrappers.cpp

Differential Revision: [D21100081](https://our.internmc.facebook.com/intern/diff/D21100081/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21100081/)!

[ghstack-poisoned]

smessmer committed:
Update on "Remove use_c10_dispatcher: unboxed_only"


All ops now do unboxing after dispatch, i.e. c10 handles unboxing and c10 registers a wrapper for the op to JIT
The last op that manually registered its own wrapper to JIT in register_aten_ops.cpp was migrated.

Since there are no ops using use_c10_dispatcher: unboxed_only anymore, we can delete the feature.

Also:
- Rename some files to more accurately describe what they're doing now:
  - OpsAlreadyMovedToC10.h/cpp -> ATenOpList.h/cpp
  - register_aten_ops.cpp -> generated_unboxing_wrappers.cpp
  - gen_jit_dispatch.py -> gen_unboxing_wrappers.cpp

Differential Revision: [D21100081](https://our.internmc.facebook.com/intern/diff/D21100081/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21100081/)!

[ghstack-poisoned]
